File: MultiplicationTrainer.java
// MultiplicationTrainer.java - Тренажёр таблицы умножения на Java (CLI)
import java.util.Scanner;
import java.util.Random;

public class MultiplicationTrainer {
    private static Scanner scanner = new Scanner(System.in);
    private static Random random = new Random();

    public static int getDifficulty() {
        System.out.println("Выберите сложность:");
        System.out.println("1. Лёгкая (1-5)");
        System.out.println("2. Средняя (1-10)");
        System.out.println("3. Сложная (1-20)");
        while (true) {
            System.out.print("Ваш выбор (1-3): ");
            try {
                int choice = Integer.parseInt(scanner.nextLine());
                if (choice == 1) return 5;
                if (choice == 2) return 10;
                if (choice == 3) return 20;
                System.out.println("Пожалуйста, введите 1, 2 или 3.");
            } catch (NumberFormatException e) {
                System.out.println("Введите число.");
            }
        }
    }

    public static int getQuestionCount() {
        while (true) {
            System.out.print("Сколько примеров вы хотите решить? (1-50): ");
            try {
                int count = Integer.parseInt(scanner.nextLine());
                if (count >= 1 && count <= 50) return count;
                System.out.println("Пожалуйста, введите число от 1 до 50.");
            } catch (NumberFormatException e) {
                System.out.println("Введите число.");
            }
        }
    }

    public static void runTrainer(int maxNum, int totalQuestions) {
        int correct = 0;
        long start = System.currentTimeMillis();

        for (int i = 1; i <= totalQuestions; i++) {
            int a = random.nextInt(maxNum) + 1;
            int b = random.nextInt(maxNum) + 1;
            System.out.printf("\nПример %d/%d: %d x %d = ? ", i, totalQuestions, a, b);
            int userAnswer;
            try {
                userAnswer = Integer.parseInt(scanner.nextLine());
            } catch (NumberFormatException e) {
                System.out.println("❌ Введите число.");
                userAnswer = -1;
            }
            int correctAnswer = a * b;
            if (userAnswer == correctAnswer) {
                System.out.println("✅ Правильно!");
                correct++;
            } else {
                System.out.printf("❌ Неправильно! Правильный ответ: %d\n", correctAnswer);
            }
        }

        long end = System.currentTimeMillis();
        double elapsed = (end - start) / 1000.0;
        double percent = (correct * 100.0) / totalQuestions;
        System.out.println("\n" + "=".repeat(30));
        System.out.println("📊 РЕЗУЛЬТАТЫ");
        System.out.printf("Всего примеров: %d\n", totalQuestions);
        System.out.printf("Правильных ответов: %d\n", correct);
        System.out.printf("Неправильных ответов: %d\n", totalQuestions - correct);
        System.out.printf("Процент правильных: %.1f%%\n", percent);
        System.out.printf("Время: %.2f сек.\n", elapsed);
        System.out.println("=".repeat(30));
    }

    public static void main(String[] args) {
        System.out.println("🧮 Тренажёр таблицы умножения");
        int maxNum = getDifficulty();
        int total = getQuestionCount();
        runTrainer(maxNum, total);
        System.out.print("Хотите повторить? (y/n): ");
        String again = scanner.nextLine().trim().toLowerCase();
        if (again.equals("y")) {
            main(args);
        } else {
            System.out.println("До свидания!");
        }
    }
}

File: multiplication_trainer.js
// multiplication_trainer.js - Тренажёр таблицы умножения на JavaScript (Node.js CLI)
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const prompt = (query) => new Promise(resolve => rl.question(query, resolve));

async function getDifficulty() {
    console.log("Выберите сложность:");
    console.log("1. Лёгкая (1-5)");
    console.log("2. Средняя (1-10)");
    console.log("3. Сложная (1-20)");
    while (true) {
        const choice = await prompt("Ваш выбор (1-3): ");
        const num = parseInt(choice);
        if (num === 1) return 5;
        if (num === 2) return 10;
        if (num === 3) return 20;
        console.log("Пожалуйста, введите 1, 2 или 3.");
    }
}

async function getQuestionCount() {
    while (true) {
        const input = await prompt("Сколько примеров вы хотите решить? (1-50): ");
        const count = parseInt(input);
        if (count >= 1 && count <= 50) return count;
        console.log("Пожалуйста, введите число от 1 до 50.");
    }
}

async function runTrainer(maxNum, totalQuestions) {
    let correct = 0;
    const start = Date.now();
    for (let i = 1; i <= totalQuestions; i++) {
        const a = Math.floor(Math.random() * maxNum) + 1;
        const b = Math.floor(Math.random() * maxNum) + 1;
        const answer = a * b;
        const userAnswer = await prompt(`\nПример ${i}/${totalQuestions}: ${a} x ${b} = ? `);
        const parsed = parseInt(userAnswer);
        if (parsed === answer) {
            console.log("✅ Правильно!");
            correct++;
        } else {
            console.log(`❌ Неправильно! Правильный ответ: ${answer}`);
        }
    }
    const elapsed = (Date.now() - start) / 1000;
    const percent = (correct / totalQuestions) * 100;
    console.log("\n" + "=".repeat(30));
    console.log("📊 РЕЗУЛЬТАТЫ");
    console.log(`Всего примеров: ${totalQuestions}`);
    console.log(`Правильных ответов: ${correct}`);
    console.log(`Неправильных ответов: ${totalQuestions - correct}`);
    console.log(`Процент правильных: ${percent.toFixed(1)}%`);
    console.log(`Время: ${elapsed.toFixed(2)} сек.`);
    console.log("=".repeat(30));
}

async function main() {
    console.log("🧮 Тренажёр таблицы умножения");
    const maxNum = await getDifficulty();
    const total = await getQuestionCount();
    await runTrainer(maxNum, total);
    const again = await prompt("Хотите повторить? (y/n): ");
    if (again.toLowerCase() === 'y') {
        await main();
    } else {
        console.log("До свидания!");
        rl.close();
    }
}

main().catch(console.error);
